flake8_trio/visitors/helpers.py

```diff
@@ -181,20 +167,23 @@ def iter_guaranteed_once_cst(iterable: cst.BaseExpression) -> bool:
             func=m.Name("range"),
         ),
     ):
-        try:
-            return (
-                len(
-                    range(
-                        *[
-                            cst_literal_eval(a.value)
-                            for a in ensure_type(iterable, cst.Call).args
-                        ]
-                    )
-                )
-                > 0
-            )
-        except Exception:  # noqa: PIE786
+        values: list[int] = []
+        for arg_arg in ensure_type(iterable, cst.Call).args:
+            arg = arg_arg.value
+            if isinstance(arg, cst.UnaryOperation):
+                if not isinstance(arg.expression, cst.Integer):
+                    return False
+                value = arg.expression.evaluated_value
+                if isinstance(arg.operator, cst.Minus):
+                    value = -value
```

Comment:
Missing the `~` operator!  https://docs.python.org/3/reference/expressions.html#unary-arithmetic-and-bitwise-operations

Reply:
Good catch! Although I suspect it's quite rare to invert bits in range loops :sweat_smile:

Reply:
And of course this checker is gonna reach that scale :grin: